internal/worker/dbaccessor/worker_namespace_test.go

```diff
@@ -160,73 +154,73 @@ func (s *namespaceSuite) TestEnsureNamespaceForModelLoopbackPreferred(c *gc.C) {
 	dbw := s.startWorker(c, ctx)
 	defer workertest.DirtyKill(c, dbw)
 
-	err := dbw.ensureNamespace("foo")
+	err := dbw.ensureNamespace(s.modelUUID.String())
 	c.Assert(err, jc.ErrorIsNil)
 
 	workertest.CleanKill(c, dbw)
 }
 
-func (s *namespaceSuite) TestEnsureNamespaceForModelWithCache(c *gc.C) {
-	defer s.setupMocks(c).Finish()
-
-	s.expectClock()
-
-	dataDir := c.MkDir()
-	mgrExp := s.nodeManager.EXPECT()
-	mgrExp.EnsureDataDir().Return(dataDir, nil).MinTimes(1)
-
-	// If this is an existing node, we do not
-	// invoke the address or cluster options.
-	mgrExp.IsExistingNode().Return(true, nil)
-	mgrExp.IsLoopbackPreferred().Return(false)
-	mgrExp.IsLoopbackBound(gomock.Any()).Return(true, nil).Times(2)
-	mgrExp.WithLogFuncOption().Return(nil)
-	mgrExp.WithTracingOption().Return(nil)
-
-	s.client.EXPECT().Cluster(gomock.Any()).Return(nil, nil)
-
-	s.expectNodeStartupAndShutdown()
-	s.expectNoConfigChanges()
-	s.clusterConfig.EXPECT().DBBindAddresses().Return(nil, errors.New("simulates absent config for initial check"))
-
-	trackedWorkerDB := newWorkerTrackedDB(s.TxnRunner())
-
-	w := s.newWorkerWithDB(c, trackedWorkerDB)
-	defer workertest.DirtyKill(c, w)
-
-	ctx, cancel := context.WithTimeout(context.Background(), testing.LongWait)
-	defer cancel()
-
-	var attempt int
-	err := s.TxnRunner().StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		attempt++
-
-		stmt := "INSERT INTO model_list (uuid) VALUES (?);"
-		result, err := tx.ExecContext(ctx, stmt, "foo")
-		c.Assert(err, jc.ErrorIsNil)
-
-		num, err := result.RowsAffected()
-		c.Assert(err, jc.ErrorIsNil)
-		c.Assert(num, gc.Equals, int64(1))
-
-		return nil
-	})
-	c.Assert(err, jc.ErrorIsNil)
-
-	dbw := w.(*dbWorker)
-	ensureStartup(c, dbw)
-
-	err = dbw.ensureNamespace("foo")
-	c.Assert(err, jc.ErrorIsNil)
-
-	// The second query will be cached.
-	err = dbw.ensureNamespace("foo")
-	c.Assert(err, jc.ErrorIsNil)
-
-	c.Assert(attempt, gc.Equals, 1)
-
-	workertest.CleanKill(c, w)
-}
+//func (s *namespaceSuite) TestEnsureNamespaceForModelWithCache(c *gc.C) {
```

Comment:
Fix me

Reply:
I need to talk to @SimonRichardson about this one.

Reply:
And Fixed